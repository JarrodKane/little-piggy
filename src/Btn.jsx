import React from "react";
import styled from "styled-components";
import { motion } from "framer-motion";

const Button = styled(motion.button)`
  font-size: 1em;
  margin: 1em;
  padding: 1em 0.5em 1em;
  background-color: white;
  border: 2px solid;
  font-weight: bold;
  text-align: center;
`;

const RollBtn = (props) => {
  const { btnAction, isGameWon, name } = props;

  return (
    <Button
      whileHover={{ scale: 1.1 }}
      whileTap={{ scale: 0.9 }}
      onClick={btnAction}
      disabled={isGameWon}
    >
      {name}
    </Button>
  );
};

export default RollBtn;
