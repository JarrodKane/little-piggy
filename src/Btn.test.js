import React from 'react';
import { render } from '@testing-library/react';

import Btn from './Btn';

test('<Btn>', () => {
  const { container } = render(<Btn />);
  //snapshot testing
  expect(container.firstChild).toMatchSnapshot();
});
