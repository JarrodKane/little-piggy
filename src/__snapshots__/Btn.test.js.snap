// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Btn> 1`] = `
<button
  class="sc-AxjAm eJNnkM"
/>
`;
